pyVHDLParser/Token/Parser.py

```diff
@@ -365,9 +359,41 @@ def GetVHDLTokenizer(cls, iterable: Iterator[str]):
 			elif tokenKind is cls.TokenKind.PossibleStringLiteralStart:
 				buffer += char
 				if char == "\"":
-					previousToken = StringLiteralToken(previousToken, buffer, start, SourceCodePosition(row, column, absolute))
+					tokenKind = cls.TokenKind.PossibleStringLiteralEnd
+
+			# State: PossibleStringLiteralEnd
+			elif tokenKind is cls.TokenKind.PossibleStringLiteralEnd:
+				if char == "\"":
+					buffer += char
+					tokenKind = cls.TokenKind.PossibleStringLiteralStart
+				else:
+					previousToken = StringLiteralToken(previousToken, buffer, start, SourceCodePosition(row, column-1, absolute-1))
 					yield previousToken
-					tokenKind = cls.TokenKind.OtherChars
+
+					start =   SourceCodePosition(row, column, absolute)

```

Comment:
The following `if`-`elif` code block is now present about eight times in this function. What are your thoughts on replacing those with calls to a separate function? 

The variables `buffer`, `previousToken`, `tokenKind` etc. would have to be passed by reference in order to support `yield`ing tokens out of it. But I do not really know a simple, non-hacky way to pass enumerations by reference (other than wrapping it manually in a temporary object/dict ofc).